src/main/java/de/hysky/skyblocker/skyblock/dwarven/GlaciteColdOverlay.java

```diff
@@ -0,0 +1,70 @@
+package de.hysky.skyblocker.skyblock.dwarven;
+
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.events.HudRenderEvents;
+import de.hysky.skyblocker.utils.Utils;
+import de.hysky.skyblocker.utils.scheduler.Scheduler;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GlaciteColdOverlay {
+    private static final Identifier POWDER_SNOW_OUTLINE = new Identifier("textures/misc/powder_snow_outline.png");
+    private static final Pattern COLD_PATTERN = Pattern.compile("Cold: -(\\d+)❄");
+    private static int cold = 0;
+    private static long resetTime = System.currentTimeMillis();
+
+    public static void init() {
+        Scheduler.INSTANCE.scheduleCyclic(GlaciteColdOverlay::update, 20);
+        HudRenderEvents.AFTER_MAIN_HUD.register(GlaciteColdOverlay::render);
+        ClientReceiveMessageEvents.GAME.register(GlaciteColdOverlay::coldReset);
+    }
+
+    private static void coldReset(Text text, boolean b) {
+        String message = text.getString();
```

Comment:
I would wrap this method in a check for being in the dwarves mines so that chat messages sent while on other islands don't get checked, saves a bit of processing. I would also check for the overlay (the boolean named b here) being false to ignore action bar updates